src/util/execute-lifecycle-script.js

```diff
@@ -65,9 +65,13 @@ export async function makeEnv(
     original: [config.commandName],
   });
 
-  // add npm_package_*
   const manifest = await config.maybeReadManifest(cwd);
   if (manifest) {
+    if (manifest.scripts && Object.prototype.hasOwnProperty.call(manifest.scripts, stage)) {
+      env.npm_lifecycle_script = manifest.scripts[stage];
```

Comment:
This is already exposed as `npm_lifecycle_event` on line 50 above. Isn't that enough?

Reply:
Ah, sorry. I read the original issue now and I understand what this is doing.